### DISCUSSION TITLE
Move cert files from caddy to postal?

### DISCUSSION THREAD

**Discussion by sinnbeck** [postalserver/postal#2091] [Help with using Postal]
So I managed to get caddy to create a viable certificate to use with postal and I manually copied them over to /opt/postal/config. Sadly this gave a "permission denied" error when trying to send (in smtp logs). So after messing around I found out that the files needs to be readable by all.

How do people handle copying the newest files over and giving the correct permissions? Just a shell script running in cron or? Or is there a better way that I am missing?

**Comment by willpower232** [comment]
I'm not sure how you apply this to caddy but yeah that sounds about right. The certificates expire every few months so there will need to be some restarts of Postal to apply the updated files.

**Reply by sinnbeck** [reply]
If I update the files on the host system they should update in the container as well I would assume? But postal needs a restart to load the new files perhaps?

I wrote this script that I run every weekend now
```
cp /opt/postal/caddy-data/caddy/certificates/acme-v02.api.letsencrypt.org-directory/postal.example.com/postal.example.com.crt /opt/postal/config/smtp.crt
cp /opt/postal/caddy-data/caddy/certificates/acme-v02.api.letsencrypt.org-directory/postal.example.com/postal.example.com.key /opt/postal/config/smtp.key
chmod o+r /opt/postal/config/smtp.*
```
I have removed my domain from the code :)

**Reply by willpower232** [reply]
> postal needs a restart to load the new files

I have something similar in our certbot cron job, presumably you could schedule that on a weekly or monthly basis in your server cron with some kind of restart, it depends on your monitoring tolerances I guess
